<a href="https://colab.research.google.com/github/kensuenck/Student-Performance-Machine-Learning-Replication/blob/main/notebooks/student_performance_prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!pip install -r requirements.txt
!python replicate_student_performance.py

Dataset loaded successfully.
Shape: (395, 34)
Columns: ['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu', 'Mjob', 'Fjob', 'reason', 'guardian', 'traveltime', 'studytime', 'failures', 'schoolsup', 'famsup', 'paid', 'activities', 'nursery', 'higher', 'internet', 'romantic', 'famrel', 'freetime', 'goout', 'Dalc', 'Walc', 'health', 'absences', 'G1', 'G2', 'G3', 'pass']

Target distribution:
pass
1    0.670886
0    0.329114
Name: proportion, dtype: float64

EXPERIMENT: WITH G1/G2

Decision Tree
-------------
Accuracy : 0.886
Precision: 0.978
Recall   : 0.849
F1-score : 0.909

Classification report:
              precision    recall  f1-score   support

           0      0.758     0.962     0.847        26
           1      0.978     0.849     0.909        53

    accuracy                          0.886        79
   macro avg      0.868     0.905     0.878        79
weighted avg      0.906     0.886     0.889        79


Random Forest
-------------
Accuracy : 0.873
Pre

## 📊 Results Summary

| Experiment        | Model          | Accuracy | Precision | Recall | F1-score |
|------------------|---------------|----------|-----------|--------|----------|
| With G1/G2       | Decision Tree | 0.886    | 0.978     | 0.849  | 0.909    |
| With G1/G2       | Random Forest | 0.873    | 0.957     | 0.849  | 0.900    |
| Without G1/G2    | Decision Tree | 0.671    | 0.708     | 0.868  | 0.780    |
| Without G1/G2    | Random Forest | 0.646    | 0.676     | 0.906  | 0.774    |

## 🧠 Interpretation

- Models perform much better when prior grades (G1, G2) are included
- Accuracy drops significantly without them (~0.88 → ~0.65)
- This shows prior grades are the strongest predictors
- However, this raises fairness concerns, as these variables may reflect existing inequalities